### ISSUE TITLE
qsanguoshav2-git: Cannot mix incompatible Qt library

### ISSUE THREAD

**Issue by htfy96** [archlinuxcn/repo#312]
```
archlinuxcn/qsanguoshav2-git 6018.f76aab3-1 [installed]
➜  ~ qsanguosha 
Cannot mix incompatible Qt library (version 0x50700) with this library (version 0x50701)
/usr/bin/qsanguosha: line 3: 30614 Aborted                 (core dumped) ./QSanguosha $@
```

Failed after upgrading to Qt5.7.1.

**Comment by htfy96** [comment]
`zeal-git` also suffers from this issue.

**Comment by htfy96** [comment]
It appears to be qt5-styleplugin's issue:
> papabean commented on 2016-12-17 20:33			 
Just ran in to an issue with a couple of qt apps: mumble-git  transmission-qt would not run and produce this error: Cannot mix incompatible Qt library (version 0x50700) with this library (version 0x50701)
> Appears to have been caused by this package needing to be rebuilt against qt5.7.1. Sounds like a minor version bump is in order.
>nfnty commented on 2016-12-17 22:11			 
@papabean Thanks, fixed.

Fixed after recompiling qt5-styleplugins.

**Comment by htfy96** [comment]
Recompiling `qt5-styleplugins` from [HEAD](http://code.qt.io/cgit/qt/qtstyleplugins.git/commit/?id=600c296f4d429ffeb8203feb54efeacc2bbea0f7) only tamed `zeal` and the setup screen of `qsanguoshav2-git`. 
The game still crashes with 
```
Cannot mix incompatible Qt library (version 0x50700) with this library (version 0x50701)
/usr/bin/qsanguosha: line 3:  2227 Aborted                 (core dumped) ./QSanguosha $@
```
when clicking "start game".

GDB log:
```
Cannot mix incompatible Qt library (version 0x50700) with this library (version 0x50701)

Thread 1 "QSanguosha" received signal SIGABRT, Aborted.
0x00007ffff4cd004f in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff4cd004f in raise () at /usr/lib/libc.so.6
#1  0x00007ffff4cd147a in abort () at /usr/lib/libc.so.6
#2  0x00007ffff5b8e4f1 in  () at /usr/lib/libQt5Core.so.5
#3  0x00007ffff5dac0e5 in QObjectPrivate::QObjectPrivate(int) () at /usr/lib/libQt5Core.so.5
#4  0x00007ffff7580a21 in  () at /usr/lib/libQt5Declarative.so.5
#5  0x00007ffff7580c22 in QDeclarativeEngine::QDeclarativeEngine(QObject*) () at /usr/lib/libQt5Declarative.so.5
#6  0x0000000000709251 in  ()
#7  0x000000000054ea1d in  ()
#8  0x00000000009de4e5 in  ()
#9  0x00007ffff5dae0c9 in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#10 0x000000000048ce08 in  ()
#11 0x000000000049162f in  ()
#12 0x00000000009d0b68 in  ()
#13 0x00007ffff5dae0c9 in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#14 0x0000000000a03f5f in  ()
#15 0x0000000000719694 in  ()
#16 0x00007ffff5dae0c9 in QMetaObject::activate(QObject*, int, int, void**) () at /usr/lib/libQt5Core.so.5
#17 0x00007ffff62a6125 in  () at /usr/lib/libQt5Network.so.5
#18 0x00007ffff62a61dc in  () at /usr/lib/libQt5Network.so.5
#19 0x00007ffff62b0fe1 in  () at /usr/lib/libQt5Network.so.5
#20 0x00007ffff6dc435c in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#21 0x00007ffff6dcbad1 in QApplication::notify(QObject*, QEvent*) () at /usr/lib/libQt5Widgets.so.5
#22 0x00007ffff5d828e0 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /usr/lib/libQt5Core.so.5
#23 0x00007ffff5dd776d in  () at /usr/lib/libQt5Core.so.5
#24 0x00007ffff1e43587 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#25 0x00007ffff1e437f0 in  () at /usr/lib/libglib-2.0.so.0
#26 0x00007ffff1e4389c in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#27 0x00007ffff5dd72bf in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#28 0x00007ffff5d80d3a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /usr/lib/libQt5Core.so.5
#29 0x00007ffff5d8923c in QCoreApplication::exec() () at /usr/lib/libQt5Core.so.5
#30 0x0000000000487145 in  ()
#31 0x00007ffff4cbd291 in __libc_start_main () at /usr/lib/libc.so.6
#32 0x00000000004895f9 in  ()
(gdb) info threads 
  Id   Target Id         Frame 
* 1    Thread 0x7ffff7f06f80 (LWP 2772) "QSanguosha" 0x00007ffff4cd004f in raise () from /usr/lib/libc.so.6
  2    Thread 0x7fffe8de5700 (LWP 2776) "QXcbEventReader" 0x00007ffff4d7c48d in poll () from /usr/lib/libc.so.6
  3    Thread 0x7fffde61a700 (LWP 2777) "QDBusConnection" 0x00007ffff4d7c48d in poll () from /usr/lib/libc.so.6
  4    Thread 0x7fffdd14c700 (LWP 2778) "gmain" 0x00007ffff4d7c48d in poll () from /usr/lib/libc.so.6
  5    Thread 0x7fffdc94b700 (LWP 2779) "gdbus" 0x00007ffff4d7c48d in poll () from /usr/lib/libc.so.6
  6    Thread 0x7fffcffff700 (LWP 2780) "pool" 0x00007ffff4d80f19 in syscall () from /usr/lib/libc.so.6
  7    Thread 0x7fffce573700 (LWP 2783) "threaded-ml" 0x00007ffff4d7c48d in poll () from /usr/lib/libc.so.6
  8    Thread 0x7fffe812f700 (LWP 2784) "QSanguosha" 0x00007ffff58eb0f7 in pthread_cond_wait@@GLIBC_2.3.2 () from /usr/lib/libpthread.so.0
  9    Thread 0x7fffe8123700 (LWP 2785) "QSanguosha" 0x00007ffff4d54ffd in nanosleep () from /usr/lib/libc.so.6
  10   Thread 0x7fffb3cb7700 (LWP 2788) "Qt bearer threa" 0x00007ffff4d7c48d in poll () from /usr/lib/libc.so.6
```

**Comment by petronny** [comment]
看起来还是各个qt库编译时使用的qt版本不一致的问题
请检查你的其他qt5库是否需要重新编译，类似styleplugins

**Comment by farseerfc** [comment]
I tried qsanguoshav2 and zeal-git but cannot reproduce the problem.
@htfy96 is your system Arch Linux (not downstream distro)? have you update to latest packages?

**Comment by farseerfc** [comment]
Sorry, it seems it indeed need a rebuild, but the current source code cannot build due to API changes.
I will try to debug and fix with upstream.

**Comment by farseerfc** [comment]
fixed in 40e63d3b7ef5bd755a447f68e49c9bf55cc7af06 and 76b768b67ff35497042b923d9c19dbafd3b88216

**Comment by htfy96** [comment]
Fixed in 6045.11a49f96-1.

**Comment by htfy96** [comment]
Broken again in Qt 5.8.

**Comment by farseerfc** [comment]
fixed in f4fb4609c0d782a3735775d43b6afe4c6bf7d27e

**Comment by htfy96** [comment]
Broken again in Qt 5.9.
